src/power10.rs

```diff
@@ -0,0 +1,125 @@
+use Integer;
+
+/// Provides methods to compute functions related to powers of 10.
+pub trait Power10: Integer {
+
+    /// Returns `true` if the number is a power of 10.
+    ///
+    /// # Examples
+    ///
+    /// ~~~
+    /// use num_integer::Power10;
+    /// assert_eq!(100u32.is_power_of_ten(), true);
+    /// assert_eq!(4u32.is_power_of_ten(), false);
+    /// ~~~
+    fn is_power_of_ten(&self) -> bool;
+}
+
+/// Returns `true` if the number is a power of 10.
+#[inline]
+pub fn is_power_of_ten<T: Power10>(x: T) -> bool {
+    x.is_power_of_ten()
+}
+
+// implementation note: the is_power_of_ten algorithm is based on a perfect hash setup with very simple hash functions
+// hash functions only use 32-bit operations for portable-speed
+// this approach is about 20% faster than using leading_zeros() (which is used for fast log operations)
+static POWER10_HASH_U16: [u32;8] = [1, 10, 10000, 0, 100, 1000, 0, 0];
+static POWER10_HASH_U32: [u32;16] = [10000, 1, 10000000, 0, 100, 0, 100000, 100000000, 1000, 0, 10, 1000000000, 0, 1000000, 0, 0];
+static POWER10_HASH_U64: [u64;32] = [1, 100000000000, 10000000000000000, 0, 1000000, 10, 0, 10000000000000, 10000000,
+    0, 1000, 0, 0, 0, 1000000000000, 0, 1000000000000000, 0, 100000000000000,
+    1000000000000000000, 100, 100000000, 100000000000000000, 0, 10000000000000000000,
+    10000, 100000, 0, 1000000000, 0, 10000000000, 0];
+
+// implementation note: reverse search is a bit faster than hash lookup for u8
+#[inline]
+fn is_pow10_u8(v: u8) -> bool {
+    if v >= 100 { return v == 100; }
+    if v >= 10 { return v == 10; }
+    if v >= 1 { return v == 1; }
+    false
```

Comment:
I think these last two lines could be collapsed to just `v == 1`.
(but LLVM might figure that out anyway...)

Reply:
Yub.